manticore/utils/log.py

```diff
@@ -11,70 +11,95 @@ def filter(self, record):
             record.stateid = '[%d]' % self.stateid
         else:
             record.stateid = ''
+
+        # Summarize logger name
+        components = record.name.split('.')
+        name = '.'.join(c[0] for c in components[:-1]) + '.' + components[-1]
+        record.name = name
+
         return True
 
-loggers = ['MANTICORE',
-           'VISITOR',
-           'EXECUTOR',
-           'CPU',
-           'REGISTERS',
-           'SMT',
-           'MEMORY',
-           'PLATFORM']
 manticore_verbosity = 0
+all_loggers = []
 
 def init_logging():
+    global all_loggers
+    all_loggers = logging.getLogger().manager.loggerDict.keys()
+        
     ctxfilter = ContextFilter()
     logfmt = ("%(asctime)s: [%(process)d]%(stateid)s %(name)s:%(levelname)s:"
               " %(message)s")
     logging.basicConfig(format=logfmt, stream=sys.stdout, level=logging.ERROR)
-    for l in loggers:
-        logging.getLogger(l).setLevel(logging.WARNING)
-        logging.getLogger(l).addFilter(ctxfilter)
-        logging.getLogger(l).setState = types.MethodType(loggerSetState,
-                                                         logging.getLogger(l))
+    for name in logging.getLogger().manager.loggerDict.keys():
+        logger = logging.getLogger(name)
+        if not name.startswith('manticore'):
+            next
+        logger.setLevel(logging.WARNING)
+        logger.addFilter(ctxfilter)
+        logger.setState = types.MethodType(loggerSetState, logger)
 
 def loggerSetState(logger, stateid):
     logger.filters[0].stateid = stateid
 
 def set_verbosity(setting):
-    global manticore_verbosity
-    zero = map(lambda x: (x, logging.WARNING), loggers)
+    global manticore_verbosity, all_loggers
```

Comment:
How would you feel about `all_loggers` being a property of `Manticore` instead of a global? That seems handy if we'd ever like to e.g. change which loggers there are at times other than initialization (like a plugin might?).

Reply:
using `logging.getLogger` is package-scoped, not Manticore scoped. Moving logging to Manticore will require changing how we log in general. Not saying it's a bad idea, just far outside the scope of this PR.